crates/sui-indexer/src/models/objects.rs

```diff
@@ -118,9 +133,161 @@ impl Object {
                 .expect("Expect the object type to be non-empty")
                 .to_string(),
             object_status: *status,
+            has_public_transfer,
             bcs,
         }
     }
+
+    pub fn to_object_response(
+        self,
+        options: &SuiObjectDataOptions,
+    ) -> Result<SuiObjectResponse, IndexerError> {
+        if self.object_type.as_str() == PACKAGE {
+            Err(IndexerError::NotImplementedError(
+                "Move package read from objects table is not implemented yet.".into(),
+            ))
+        } else {
+            let object_id = self.object_id.parse().map_err(|e| {
+                IndexerError::SerdeError(format!(
+                    "Failed to parse object id: {}, error: {}",
+                    self.object_id, e
+                ))
+            })?;
+            let digest = self.object_digest.parse().map_err(|e| {
+                IndexerError::SerdeError(format!(
+                    "Failed to parse object digest: {}, error: {}",
+                    self.object_digest, e
+                ))
+            })?;
+            let previous_transaction = self.previous_transaction.parse().map_err(|e| {
+                IndexerError::SerdeError(format!(
+                    "Failed to parse previous transaction: {}, error: {}",
+                    self.previous_transaction, e
+                ))
+            })?;
+
+            // only one BCS for non-package object
+            let obj_bcs = self
+                .bcs
+                .first()
+                .cloned()
+                .expect("Expect the BCS data to be non-empty")
+                .1;
+            let struct_tag = StructTag::from_str(&self.object_type).map_err(|e| {
+                IndexerError::SerdeError(format!("Failed to parse struct tag: {}", e))
+            })?;
+
+            let move_obj = unsafe {
+                MoveObject::new_from_execution_with_limit(
+                    struct_tag.clone(),
+                    self.has_public_transfer,
+                    SequenceNumber::from_u64(self.version as u64),
+                    obj_bcs,
+                    u64::MAX,
+                )
+                .map_err(|e| {
+                    IndexerError::SerdeError(format!("Failed to parse move object: {}", e))
+                })
+            }?;
+
+            let owner = self.get_owner()?;
+            let obj = SuiBaseObject::new_move(move_obj.clone(), owner, previous_transaction);
+            let storage_rebate = obj.storage_rebate;
```

Comment:
storage_rebate is always needed, which is derived from obj, as a result `owner`, `previous_transaction` and `move_obj` always need to be parsed, even though they are not rendered.

Reply:
some optimizations can be done here for less parsing.